dv360proxy/api.js

```diff
@@ -1,195 +1,66 @@
-            }
-            console.error(
-                `${requestId}: DV360API.createQuery(${JSON.stringify(query)}) failed with ${errMessage}`
-            );
-            throw err;
-        }
+        return await dbmApi.createQuery(requestId, query);
+    };
+
+    this.createQuery_v2 = async ({query}) => {
+        return await dbmApiV2.createQuery(requestId, query);
     };
 
     this.runQuery = async ({queryId, data}) => {
-        const url = `/query/${queryId}`;
-        try {
-            const queriesResponse = await dbmAPI.post(url, data);
-            return queriesResponse.data;
-        } catch (err) {
-            let errMessage = err.stack || err.toString();
-            if (err.response) {
-                errMessage = JSON.stringify(err.response.data);
-            }
-            console.error(
-                `${requestId}: DV360API.runQuery(${JSON.stringify(queryId)}) failed with ${errMessage}`
-            );
-            throw err;
-        }
+        return await dbmApi.runQuery(requestId, queryId, data);
+    };
+
+    this.runQuery_v2 = async ({queryId, data}) => {
+        return await dbmApiV2.runQuery(requestId, queryId, data);
     };
 
     this.deleteQuery = async ({queryId}) => {
-        const url = `/query/${queryId}`;
-        try {
-            await dbmAPI.delete(url);
-            return true;
-        } catch (err) {
-            let errMessage = err.stack || err.toString();
-            if (err.response) {
-                errMessage = JSON.stringify(err.response.data);
-            }
-            console.error(
-                `${requestId}: DV360API.deleteQuery(${JSON.stringify(queryId)}) failed with ${errMessage}`
-            );
-            throw err;
-        }
+        return await dbmApi.deleteQuery(requestId, queryId);
+    };
+
+    this.deleteQuery_v2 = async ({queryId}) => {
+        return await dbmApiV2.deleteQuery(requestId, queryId);
     };
 
     this.getQueryReports = async ({queryId, pageToken}) => {
-        const url = `/queries/${queryId}/reports`;
-        try {
-            const queriesResponse = await dbmAPI.get(url, { params: { pageToken: pageToken } });
-            return queriesResponse.data;
-        } catch (err) {
-            let errMessage = err.stack || err.toString();
-            if (err.response) {
-                errMessage = JSON.stringify(err.response.data);
-            }
-            console.error(
-                `${requestId}: DV360API.deleteQuery(${JSON.stringify(queryId)}) failed with ${errMessage}`
-            );
-            throw err;
-        }
+        return await dbmApi.getQueryReports(requestId, queryId, pageToken);
+    };
+
+    this.getQueryReports_v2 = async ({queryId, pageToken}) => {
+        return await dbmApiV2.getQueryReports(requestId, queryId, pageToken);
     };
 
     this.ping = async () => {
+        return await ping(async () => await dbmApi.getQueries(requestId));
+    }
+
+    this.ping_v2 = async () => {
```

Comment:
do you think we should have `ping_v2` or we can just migrate `ping` to V2 API? Request and Response is not tied to DB API version, so I don't see a reason to have `ping_v2`

Reply:
Done, ping V1 is no more and ping V2 is now just ping